### ISSUE TITLE
The Russian language is entered from the keyboard only in applications from the maemo leste repository

### ISSUE THREAD

**Issue by sabelluskib** [maemo-leste/bugtracker#688]
Video: https://drive.google.com/file/d/1TqSZVthJGuPBZK4NIITkaXkNUKjTzvuF/view?usp=share_link . The video turned out to be long (5 minutes), attached a link to google drive.

Hello! I attached a video to make the problem easier to see. At the beginning of the video, I demonstrate that initially the characters on the keyboard do not correspond to reality. After enabling setxkbmap ru, the Russian layout works fine, all characters correspond to the pattern on the keyboard. But this is only in applications from the maemo leste repository.

Later it can be seen that everything works fine in leafpad. Switching languages works, a notification about changing the language appears.

At the moment of the video it is clear that the Russian language is not found at all in Geany, language switching does not work, there are no notifications, but the layout is saved! The arrows behave according to the images on the keyboard, special characters are entered as on the keyboard. By the way, this does not work regardless of which layout is before the launch. There are also problems in CLI applications from devuan, I can't enter Russian characters in vim.

The same behavior can be seen further when working with telegram-desktop. There is no language switching, no notifications, but special symbols and arrows work.

In the end, I opened the contacts application to show once again that everything works from maemo applications. I hope I was able to explain the situation. I really like the project, I really want to popularize it among the Russian-speaking community. Thanks!

**Comment by MerlijnWajer** [comment]
Could you perhaps make the video public? Thanks!

**Comment by sabelluskib** [comment]
Yes, I uploaded the video to youtube. Github does not allow you to upload videos larger than 10 MB. Link: https://www.youtube.com/watch?v=mNo6FnZNDuA

**Comment by MerlijnWajer** [comment]
Thank you for this video, it is quite clear. I think it shows that indeed applications that support "him" (Hildon Input Method) work as expected, but others do not. There are probably a few things to do for us:

1. Figure out how to set up a device so that you don't have to type `setxkbmap ru` every time when it starts.
2. Support HIM in gtk3 and qt5. This will make the majority of the applications work as expected.
3. See if we can extend HIM or change it so that it works more globally (switching keyboard layout from a non-him application being one of them). Some work was done on this, but not for all the layouts, I think.

**Comment by home14352** [comment]
Maybe we need to include setxkbmap ru into .xinitrc?
